src/cargo/ops/registry.rs

```diff
@@ -560,6 +560,9 @@ pub fn configure_http_handle(config: &Config, handle: &mut Easy) -> CargoResult<
     if let Some(proxy) = http_proxy(config)? {
         handle.proxy(&proxy)?;
     }
+    handle.proxy_auth(&http.proxy_auth.to_easy())?;
+    handle.proxy_username(http.proxy_username.as_deref().unwrap_or(""))?;
```

Comment:
Can you add a comment about why these need to be called with empty string (instead of just not calling them)?

Reply:
The comment is in the commit message [0] (second paragraph). Do you want another one in the code?

[0] <https://github.com/rust-lang/cargo/pull/11433/commits/e7a590516c58b85dfaa98bfcdfb7f77c9b551173>